sqlmesh/core/model/definition.py

```diff
@@ -1876,6 +1877,21 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
+def _resolve_custom_session_params(
+    defaults: t.Dict[str, t.Any] | None, **kwargs: t.Dict[str, t.Any]
+) -> None:
+    if kwargs.get("session_properties") and defaults and defaults.get("session_properties"):
```

Comment:
So what happens if there session properties in defaults but not in model definitions? Don't we still need to convert the dict into a sqlglot expression?